icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -769,6 +769,58 @@ MeasureUnit MeasureUnit::product(const MeasureUnit& other, UErrorCode& status) c
     return std::move(impl).build(status);
 }
 
+/**
+ * Searches the `simplifiedUnits` for a unit with the same base identifier in the `newUnit`, for example
+ * `meter` and `meter` or `millimeter` and `centimeter`.
+ * After finding a match, the matched unit in simplified unit will be merged with the `newUnit` and
+ * `true` will be returned. Otherwise, `false` will be returned.
+ */
+bool findAndSet(MaybeStackVector<MeasureUnit> &simplifiedUnits, const MeasureUnit &newUnit,
+                UErrorCode &status) {
+    for (int i = 0, n = simplifiedUnits.length(); i < n; i++) {
+        auto simplifiedUnitImpl = SingleUnitImpl::forMeasureUnit(*simplifiedUnits[i], status);
+        auto newUnitImpl = SingleUnitImpl::forMeasureUnit(newUnit, status);
+        if (simplifiedUnitImpl.identifier == newUnitImpl.identifier) {
```

Comment:
@sffc : should we bring back a "getBaseIdentifier()" function or similar? Vs having to refactor this code to use public MeasureUnit APIs?

Reply:
Maybe we need to bring back `identifier` then: we dropped it because it wasn't safe, and at the time nothing was using it. I will try to find a safe string to point it at.

Reply:
Acknowledged!

Reply:
Taking care of it: I've sent a PR for `getIdentifier()` and will use that in my merge of upstream/master.